src/v/cluster/health_monitor_types.h

```diff
@@ -71,6 +73,16 @@ struct topic_status {
     friend std::ostream& operator<<(std::ostream&, const topic_status&);
     friend bool operator==(const topic_status&, const topic_status&) = default;
 };
+
+struct node_local_state {
+    static constexpr int8_t current_version = 0;
```

Comment:
I don't think this `static constexpr int8_t current_version = 0;` is used. if you plan to expand this structure and serialize more state in the future--since it is embedded in node_health_report--it will be useful to encode and verify this nested version. if you don't plan on expanding the structure, then the version can be removed.